### PR TITLE
v0.2.1

### PR DESCRIPTION
**Pull request by dfordivam**
# Checklist

- [x] App update process has been followed 
- [x] Target branch is `develop` 
- [x] Application version has been bumped 

Changelog for v0.2.1

* Update SDK
* Allow the app to auto lock for PIN, and show screensaver.
* Make the settings persist across app restarts.

### CODE REVIEW

File: rust-app/Cargo.toml [hunk 1]
```diff
@@ -11,42 +11,39 @@ arrayvec = { version = "0.7.2", default-features = false }
...
 test = false
 path = "bin-src/main.rs"
 
-[patch."https://github.com/LedgerHQ/ledger-nanos-sdk.git".nanos_sdk]
-git = "https://github.com/alamgu/ledger-nanos-sdk.git"
-branch = "relocating-loader-w-fixes"
+[patch.crates-io]
+ledger_device_sdk = { git = "https://github.com/alamgu/ledger-nanos-sdk.git", branch = "alamgu-branch" }
+ledger_secure_sdk_sys = { git = "https://github.com/alamgu/ledger-nanos-sdk.git", branch = "alamgu-branch" }
+include_gif = { git = "https://github.com/alamgu/ledger-nanos-sdk.git", branch = "alamgu-branch" }
```
Comment (xchapron-ledger) [inline rust-app/Cargo.toml:46]: Any reason why this is still needed?

File: rust-app/nanos.json
Comment (xchapron-ledger) [inline rust-app/nanos.json:1]: any reason why you need specific target files?

**Comment by yogh333** [comment]
One CI test is failing. You need to update the database as detailed here: https://github.com/LedgerHQ/ledger-app-database/tree/main. Thanks !

**Comment by yogh333** [comment]
> One CI test is failing. You need to update the database as detailed here: https://github.com/LedgerHQ/ledger-app-database/tree/main. Thanks !

I have updated the database (see this [commit](https://github.com/LedgerHQ/ledger-app-database/commit/61c18e3c923bed01ca6ee313a26ab6adbe4a5e99) and the test passed 🥳 
You will have to do the same for other apps please 🙏

File: rust-app/Cargo.toml (hunk 1 above)
Comment (dfordivam) [reply rust-app/Cargo.toml:46]: There is only one commit necessary for `ledger_device_sdk` (https://github.com/alamgu/ledger-nanos-sdk/commit/c55d05853501b44c93c36599f83f71c0ed1e52cd). The patches for `ledger_secure_sdk_sys`, and `include_gif` were needed for making `cargo build` to work, but they are identical to the `ledger_secure_sdk_sys = "1.2.0"` and `include_gif = "1.0.1"` respectively.

Comment (dfordivam) [reply rust-app/Cargo.toml:46]: There are a one more commit, but that could be removed from this branch if required.
Here are all the diffs -> https://github.com/LedgerHQ/ledger-device-rust-sdk/compare/ledger_device_sdk%401.5.1...alamgu:ledger-nanos-sdk:alamgu-branch?expand=1

File: rust-app/nanos.json
Comment (dfordivam) [reply rust-app/nanos.json:1]: These are necessary for a few `cargo` commands like
```
cargo test --target=$DEVICE.json
cargo-clippy -- --target=$DEVICE.json
cargo stack-sizes --target=$DEVICE.json
 ```
 
These can be obtained by referring to the SDK, but having them here is generally more convenient.

**Comment by yogh333** [comment]
> > One CI test is failing. You need to update the database as detailed here: https://github.com/LedgerHQ/ledger-app-database/tree/main. Thanks !
> 
> I have updated the database (see this [commit](https://github.com/LedgerHQ/ledger-app-database/commit/61c18e3c923bed01ca6ee313a26ab6adbe4a5e99)) and the test passed 🥳 You will have to do the same for other apps please 🙏

Actually it has been also done for other apps as well ;)

**Comment by dfordivam** [comment]
> > > One CI test is failing. You need to update the database as detailed here: https://github.com/LedgerHQ/ledger-app-database/tree/main. Thanks !
> > 
> > 
> > I have updated the database (see this [commit](https://github.com/LedgerHQ/ledger-app-database/commit/61c18e3c923bed01ca6ee313a26ab6adbe4a5e99)) and the test passed 🥳 You will have to do the same for other apps please 🙏
> 
> Actually it has been also done for other apps as well ;)

Thanks @yogh333! I did notice that when I was trying to create a commit.

**Comment by yogh333** [comment]
Hello, 

 Could you please update the PR with regard to the last Rust SDK (version `1.7.1`) and delete any dependencies with your alamgu fork (and do the same for `app-pocket` and `app-kadena`) 🙏 ?

File: rust-app/nanos.json
Comment (dfordivam) [reply rust-app/nanos.json:1]: Files are removed

File: ledger_secure_sdk_sys-bindings/nanox/bindings.rs
[the remark this answers is not in the document]
Comment (dfordivam) [reply ledger_secure_sdk_sys-bindings/nanox/bindings.rs:1]: These have been removed

**Comment by dfordivam** [comment]
> Hello,
> 
> Could you please update the PR with regard to the last Rust SDK (version `1.7.1`) and delete any dependencies with your alamgu fork (and do the same for `app-pocket` and `app-kadena`) 🙏 ?

The PR now uses `1.7.1`

**Review by yogh333 — APPROVED** [review]
LGTM
Thanks

**Review by yogh333 — APPROVED** [review]

**Pull request merged**